rosidl_typesupport_cpp/resource/action__type_support.cpp.em

```diff
@@ -63,3 +63,18 @@ get_action_type_support_handle<@('::'.join([package_name] + list(interface_path.
 }
 
 }  // namespace rosidl_typesupport_cpp
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+ROSIDL_TYPESUPPORT_CPP_PUBLIC
```

Comment:
The problem is that `rosidl_typesupport_cpp` does not depend on `rosidl_generator` so I would have to create a visibility header for `rosidl_typesupport_cpp` as well.
I can do that but I'm not really sure what for since these libraries are built for dynamic loading anyway and I don't see why anyone should want this method to be anything other than always visible.
However, I'm not 100% sure I understand all use cases of visibility control so if you think that makes sense I can add it.

Reply:
In general, the PUBLIC macros are mostly there for Windows compatibility.  That is, Windows requires that when compiling a library, these macros are set to `dllexport`, and that when using the library, these macros are set to `dllimport`.

In this particular case, I think I agree with @StefanFabian in that all of the other functions in this package use `ROSIDL_TYPESUPPORT_CPP_PUBLIC`, so we can use that here as well.  But maybe @sloretz has something else in mind.